File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cur = 0, cnt = 0;

        for (int num : nums) {
            if (cnt == 0) {
                cur = num;
                cnt++;
            } else if (cur == num) {
                cnt++;
            } else {
                cnt--;
            }
        }

        return cur;
    }
}

File: 0169-majority-element/0169-majority-element.kt
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var cur = 0
        var cnt = 0

        for (num in nums) {
            if (cnt == 0) {
                cur = num
                cnt++
            } else if (cur == num) {
                cnt++
            } else {
                cnt--
            }
        }

        return cur
    }
}